File: srm196/Education.java
public class Education {

	public int minimize(String desire, int[] tests) {
		int res = 0;
		String grades = "ABCD";
		int[] points = {90, 80, 70, 60};
		int totalPoints = points[grades.indexOf(desire)] * (tests.length + 1);
		int earned = 0;
		for (int i: tests) {
			earned += i;
		}
		res = totalPoints - earned;
		if (res <= 0) {
			res = 0;
		} else if (res > 100) {
			res = -1;
		}
	return res;
	}
} 

File: srm196/education.py
class Education:
	def minimize(self, desire, tests):
		res = 0
		grades = ['A', 'B', 'C', 'D']
		points = [90, 80, 70, 60]
		total_points = points[grades.index(desire)] * (len(tests) + 1)
		res = total_points - sum(tests)
		if res <= 0: return 0
		elif res > 100: return -1
		return res 